src/psyclone/tests/alggen_test.py

```diff
@@ -162,7 +162,7 @@ def test_single_function_invoke_qr():
                                    "1.1.0_single_invoke_xyoz_qr.f90"),
                       api="dynamo0.3")
     gen = str(alg).lower()
-    assert "use testkern_qr" not in gen
+    assert "use testkern_wrong_file_name" not in gen
```

Comment:
I think there's some confusion here. We want this file to *only* be used to illustrate what happens if a file has the wrong name. For everything else we should use a valid kernel with a valid filename. (I thought PSyclone would reject such a filename anyway but apparently we don't :-( That's not something for this PR though.) Therefore, perhaps we need a `testkern_qr_mod` file and `testkern_wrong_file_name` too?